### DISCUSSION TITLE
Manually Fire on Change event after setting Input value for Combobox

### DISCUSSION THREAD

**Discussion by rubber-duck-software** [tailwindlabs/headlessui#2281] [Ideas]
Currently, the combobox programmatically sets the input value but it is difficult to detect when this happens because no event is fired. It would be useful if we manually fired a "change" event when the input is set so that users can react appropriately

**Comment by peteruithoven** [comment]
Doesn't the `Combobox.Input`'s `onChange` do that? 
I did find a related issue, when clicking the `Combobox.Button`, the Combobox is opened and the Input's value is reset. This change isn't communicated. Even when the input field remains whether the Combobox is open or not. 

My solution to prevent the Combobox from clearing the input was to control the value of `Combobox.Input`. By using `useState`, passing that to it's `value` and updating that state from the `onChange`. 

Only problem is, I can't easily take an action when the Combobox is opened like clearing the input. There is no event for that. And the `open` variable isn't usable outside of the Combobox.
